tools/runners/run-updater.js

```diff
@@ -1,60 +1,52 @@
-var Console = require('../console/console.js').Console;
+import { Console } from '../console/console';
 
-var Updater = function () {
-  var self = this;
-  self.timer = null;
-};
+const CHECK_UPDATE_INTERVAL = 3 * 60 * 60 * 1000; // every 3 hours
 
 // XXX make it take a runLog?
 // XXX need to deal with updater writing messages (bypassing old
 // stdout interception.. maybe it should be global after all..)
-Object.assign(Updater.prototype, {
-  start: function () {
-    var self = this;
+export class Updater {
+  constructor() {
+    this.timer = null;
+  }
 
-    if (self.timer) {
-      throw new Error("already running?");
+  start() {
+    if (this.timer) {
+      throw new Error('already running?');
     }
 
+    const self = this;
     // Check every 3 hours. (Should not share buildmessage state with
     // the main fiber.)
     async function check() {
       self._check();
     }
 
-    self.timer = setInterval(check, 3 * 60 * 60 * 1000);
+    this.timer = setInterval(check, CHECK_UPDATE_INTERVAL);
 
     // Also start a check now, but don't block on it. (This should
     // not share buildmessage state with the main fiber.)
     check();
-  },
+  }
 
-  _check: function () {
-    var self = this;
-    var updater = require('../packaging/updater.js');
+  _check() {
+    const updater = require('../packaging/updater');
     try {
-      updater.tryToDownloadUpdate({showBanner: true});
+      updater.tryToDownloadUpdate({ showBanner: true });
     } catch (e) {
       // oh well, this was the background. Only show errors if we are in debug
       // mode.
-      Console.debug("Error inside updater.");
+      Console.debug('Error inside updater.');
       Console.debug(e.stack);
-      return;
     }
-  },
+  }
 
-  // Returns immediately. However if an update check is currently
+  // Returns immediately. However, if an update check is currently
   // running it will complete in the background. Idempotent.
-  stop: function () {
-    var self = this;
+  stop() {
+    if (!this.timer) return;
```

Comment:
There was a bug here.

We want clearInterval only if it exists like this.timer && clearInterval(this.timer)